google/auth/_default.py

```diff
@@ -237,6 +237,7 @@ def _get_gcloud_sdk_credentials(quota_project_id=None):
     credentials, project_id = load_credentials_from_file(
         credentials_filename, quota_project_id=quota_project_id
     )
+    credentials._cred_file_path = credentials_filename
```

Comment:
nit: why does the Env variable file path explain the origin but this does not?

Reply:
this is gcloud user credential, the file path is not from env var, it is a well known location, i.e. ~./config/gcloud/application_default_credentials.json